bin/detect

```diff
@@ -1,10 +1,74 @@
 #!/usr/bin/env bash
 # bin/detect <build-dir>
 
+error() {
+  local c="2,999 s/^/ !     /"
+	# send all of our output to stderr
+	exec 1>&2
+
+	echo -e "\033[1;31m" # bold; red
+	echo -n " !     ERROR: "
+	# this will be fed from stdin
+  case $(uname) in
+		Darwin) sed -l "$c";; # mac/bsd sed: -l buffers on line boundaries
+		*)      sed -u "$c";; # unix/gnu sed: -u unbuffered (arbitrary) chunks of data
+	esac
+	echo -e "\033[0m" # reset style
+	exit 1
+}
+
 if [ -f $1/package.json ]; then
   echo 'Node.js'
   exit 0
 fi
 
->&2 echo 'Node.js: package.json not found in application root'
+if [[ -f "$1/.slugignore" ]] && grep -Fxq "package.json" "$1/.slugignore"; then
```

Comment:
Have you ever seen this in the wild where someone put their package.json in their .slugignore?

Reply:
Hmmm... This is already merged into master and shouldn't be in this PR. I'll rebase.

This is the relevant PR: https://github.com/heroku/heroku-buildpack-nodejs/pull/575 and yes, it came out of a support ticket for that exact thing